src/main/webapp/app/exercises/programming/manage/grading/charts/test-case-distribution-chart.component.ts

```diff
@@ -41,6 +41,9 @@ export class TestCaseDistributionChartComponent implements OnChanges {
     pointsChartDatasets: ChartDataSets[] = [];
 
     ngOnChanges(): void {
+        if (this.testCases == undefined) {
+            this.testCases = [];
```

Comment:
I think we could check for undefined with the ? operator right? 
so basically `this.testCases?.filter...`

Reply:
`this.testCases = this.testCases?.filter` would still set `this.testCases` to null or undefined again, which will then cause problems in the rest of this function. And I think this solution is cleaner for a quick fix before we rewrite all of that.

Reply:
Yes, I thought about optional chaining as well, but since this.testCases is used in several places I also thought that a check at the beginning should be better, as @FrankeLukas mentioned.

Reply:
okay sounds reasonable!

Reply:
👍